src/components/Accordion/index.tsx

```diff
@@ -17,21 +17,46 @@ export interface AccordionProps {
   className?: string;
   /** Unique id passed to data-qa-id property of the outermost wrapper of the component. Used for automation testing  */
   dataQa?: string | null;
+  /** Whether to have no padding in the accordion content */
   noPadding?: boolean;
+  /** Whether to always show the toggler icon or only show it when accordion is opened or hovered */
+  alwaysShowToggler?: boolean;
+  /** If true, the component expands or collapsed only when the toggler icon is clicked */
+  openOnClickTogglerOnly?: boolean;
 }
 
-export const Accordion = ({ title, className, children, dataQa, noPadding = false }: AccordionProps) => {
+export const Accordion = ({
+  title,
+  className,
+  children,
+  dataQa,
+  noPadding = false,
+  alwaysShowToggler = true,
+  openOnClickTogglerOnly = false,
+}: AccordionProps) => {
   const [isOpen, toggleAccordion] = React.useState(false);
+
+  const onClickHead = () => {
+    if (openOnClickTogglerOnly) return;
+    toggleAccordion(!isOpen);
+  };
+
+  const onClickToggler = () => {
+    if (!openOnClickTogglerOnly) return;
+    toggleAccordion(!isOpen);
+  };
+
   return (
     <div className={classNames("accordion", className)} data-qa-id={dataQa}>
-      <div className={classNames("accordionHead", { isOpen })} onClick={() => toggleAccordion(!isOpen)}>
-        <div className="accordionTitle">{title}</div>{" "}
+      <div className={classNames("accordionHead", { isOpen, "show-toggler": alwaysShowToggler })} onClick={onClickHead}>
+        <div className="accordionTitle">{title}</div>
         <i
           className={classNames("accordionToggler", { "fa-icon-chevron-down": !isOpen, "fa-icon-chevron-up": isOpen })}
+          onClick={onClickToggler}
         />
       </div>
       <div className={classNames("accordionBody", { show: isOpen })}>
-        <div className={classNames("accordionContent", {noPadding})}>{children}</div>
+        <div className={classNames("accordionContent", { noPadding })}>{children}</div>
```

Comment:
is the styling of `noPadding` already handled?

Reply:
yes